Updating drafting process to include test plans (#25256)
fleetdm/fleet: handbook/company/product-groups.md

- **Scope transparency:** Does this change the scope of access that Fleet has on end user devices?  If so, describe this user story so that it includes the edits necessary to the [transparency guide](https://fleetdm.com/transparency).
- **Measurement?:** User stories are small changes that are best served by being released as quickly as possible in order to get real world feedback, whether quantitative or qualitative.  The norm is NOT to prioritize additional analytics or measurement work.  Is it especially important for the change described by this user story to come with extra investment in measuring usage, adoption, and success?  If so, describe what measurements we need to implement, along with the current state of any existing, related measurements.
- **QA:** Changes are tested by hand prior to submitting pull requests. In addition, quality assurance will do an extra QA check prior to considering this story "done".  Any special QA notes?

> - **Follow-through:** Is there anything in particular that we should inform others (people who aren't in this product group) about after this user story is released?  For example: communication to specific customers, tips on how best to highlight this in a release post, gotchas, etc.

Author: @noahtalerman I'm suggesting removing this from the handbook because anything needed for the user story to be fully drafted should be list in the user story template.

Reviewer: Agreed. Good spot to de-duplicate and remove content.

#### Defining "done"

To successfully deliver a user story, the people working on it need to know what "done" means. 

> Every user story has a product and engineering checklist that is completed before the user story is estimated. This populates the user story with the requirements, wireframes, and test plan necessary for the product group to effectively specify, estimate, implement, and test the change. The Product Designer is the DRI for completing the product checklist, and the Engineering Manager (EM) is the DRI for comleting the engineering checklist.

Author: ```suggestion
Every user story has a product and engineering checklist that is completed before the user story is estimated. This populates the user story with the requirements, wireframes, and test plan necessary for the product group to effectively specify, estimate, implement, and test the change. The Product Designer is the DRI for completing the product checklist, and the Engineering Manager (EM) is the DRI for completing the engineering checklist.
```

> When the Product Designer has completed the product checklist, it is moved to the "In review" column of the drafting board and reviewed during the (weekly user story review)[https://fleetdm.com/handbook/company/product-groups#user-story-reviews] rituals.

Reviewer: ```suggestion
When the Product Designer has completed the product checklist, it is moved to the "In review" column of the drafting board and reviewed during the [weekly user story review](https://fleetdm.com/handbook/company/product-groups#user-story-reviews) rituals.
```

Quick link formatting fix

### Design reviews

> Design reviews are conducted daily between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and contributors (most often Product Designers) proposing changes to Fleet's interfaces, such as the graphical user interface (GUI) or REST API.  This fast cadence shortens the feedback loop, makes progress visible, and encourages early feedback. This helps Fleet stay intentional about how the product is designed and minimize common issues like UI inconsistencies or accidental breaking changes to the API.

Author: @noahtalerman I'm removing myself since I haven't been attending. Feel free to leave me as optional, or remove me.

- Bring 1 key engineer who has been helping out with the user story, when possible and helpful.
- Read Fleet's [best practices for meetings](https://fleetdm.com/handbook/company/communications#meetings).

### User story reviews

User story reviews [happen weekly](https://fleetdm.com/handbook/product-design#rituals) between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and the each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "In review" column. The Product Designer is the DRI for completing all product checklist items before bringing to review.

The purpose of the review is to familiarize the EM and QA Engineer with the user story, and provide an opportunity to ask questions, clarify requirements, and highlight potential implementation issues. The first draft of the test plan produced by the Product Designer is reviewed and revised as needed during the call. The QA Engineer is the DRI for finalizing the test plan.

> If there are design changes, the user story is brought back into product design for additional drafting. If there are no design changes, the story is assigned to the EM to [complete the drafting process](#defining-done) before bringing to estimation.

Author: ```suggestion
The purpose of the user story review is to align product, engineering, and QA on functionality and implementation details. Wireframe reviews occur daily during [design reviews](https://fleetdm.com/handbook/company/product-groups#design-reviews) where contributors are welcome to join and provide design feedback in the agenda document. However, sometimes there are design changes needed if a gap is discovered or an implementation issue is raised during user story review. If there are design changes, the user story is moved back to the "In progress" column for additional drafting. If there are no design changes, the story is assigned to the EM to [complete the drafting process](#defining-done) before bringing to estimation.
```

- Bring 1 key engineer who has been helping out with the user story, when possible and helpful.
- Read Fleet's [best practices for meetings](https://fleetdm.com/handbook/company/communications#meetings).

### User story reviews

> User story reviews happen weekly between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "Ready so spec" column. The Product Designer is the DRI for completing all "Product" checklist items before bringing to review. 

Author: ```suggestion
User story reviews happen weekly between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "Ready to spec" column. The Product Designer is the DRI for completing all "Product" checklist items before bringing to review. 
```

- Bring 1 key engineer who has been helping out with the user story, when possible and helpful.
- Read Fleet's [best practices for meetings](https://fleetdm.com/handbook/company/communications#meetings).

### User story reviews

> User story reviews (happen weekly)[https://fleetdm.com/handbook/product-design#rituals] between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and the each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "In review" column. The Product Designer is the DRI for completing all product checklist items before bringing to review.

Reviewer: ```suggestion
User story reviews [happen weekly](https://fleetdm.com/handbook/product-design#rituals) between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and the each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "In review" column. The Product Designer is the DRI for completing all product checklist items before bringing to review.
```
Link formatting fix.

> User story reviews (happen weekly)[https://fleetdm.com/handbook/product-design#rituals] between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and the each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "Ready to spec" column. The Product Designer is the DRI for completing all product checklist items before bringing to review.

Author: ```suggestion
User story reviews (happen weekly)[https://fleetdm.com/handbook/product-design#rituals] between the [Head of Product Design](https://fleetdm.com/handbook/product-design#team) and the each product group's Product Designer, Engineering Manager (EM), and Quality Assurance (QA) Engineer. The attendees review all user stories that have completed product design in the past week and are in the "In review" column. The Product Designer is the DRI for completing all product checklist items before bringing to review.
```

The purpose of the review is to familiarize the EM and QA Engineer with the user story, and provide an opportunity to ask questions, clarify requirements, and highlight potential implementation issues. The first draft of the test plan produced by the Product Designer is reviewed and revised as needed during the call. The QA Engineer is the DRI for finalizing the test plan.

> If there are design changes, the user story is brought back into product design for additional drafting. If there are no design changes, the story is assigned to the EM to [complete the drafting process](#defining-done) before bringing to estimation.

Reviewer: ```suggestion
Suggesting changes to the wireframes is not the purpose of the review. This happens during [design review](https://fleetdm.com/handbook/product-design#design-reviews). However, sometimes there are design changes needed if a gap is discovered or an implementation issue is raised. If there are design changes, the user story is moved back to the "In progress" column for additional drafting. If there are no design changes, the story is assigned to the EM to [complete the drafting process](#defining-done) before bringing to estimation.
```

I think we want to be explicit that proposing changes doesn't happen during this call. That happens during design review. My understanding is that the "user story review" call is about reviewing the test plan alongside the designs to see if we're ready to finalize the test plan.

Reviewer: Also, heads up that I removed the "Design consultation" section: https://fleetdm.com/handbook/company/product-groups#design-consultation

I'm not sure what this call is. I don't think it needs to be handbooked.

Author: I'll revise with slightly different wording. 

Re: Design consultation, no problem, those can just happen ad-hoc as needed. Our goal s that the teams are discussing these things proactively after standup if/when needed and not needing more meetings.
